drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_common.c

```diff
@@ -126,61 +293,47 @@ int spi_mcux_configure(const struct device *dev, const struct spi_config *spi_cf
-		 * CR register cannot be written.
-		 */
-		LPSPI_Enable(base, false);
-		while ((base->CR & LPSPI_CR_MEN_MASK) != 0U) {
-			/* Wait until LPSPI is disabled. Datasheet:
-			 * After writing 0, MEN (Module Enable) remains set until the LPSPI has
-			 * completed the current transfer and is idle.
-			 */
-		}
+	/* specific driver implementation should set up watermarks and interrupts.
+	 * we reset them here to avoid any unexpected events during configuring.
+	 */
+	base->FCR = 0;
+	base->IER = 0;
 
-		/* this is workaround for ERR050456 */
-		base->CR |= LPSPI_CR_RST_MASK;
-		base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;
-		base->CR = 0x00U;
+	/* this is workaround for ERR050456 */
+	base->CR |= LPSPI_CR_RST_MASK;
+	base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;
+
+	/* Setting the baud rate requires module to be disabled. */
+	base->CR = 0;
+	while ((base->CR & LPSPI_CR_MEN_MASK) != 0) {
+		/* According to datasheet, should wait for this MEN bit to clear once idle */
 	}
 
 	data->ctx.config = spi_cfg;
 
-	LPSPI_MasterGetDefaultConfig(&master_config);
-
-	master_config.bitsPerFrame = word_size < 8 ? 8 : word_size; /* minimum FRAMSZ is 8 */
-	master_config.cpol = (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPOL)
-				     ? kLPSPI_ClockPolarityActiveLow
-				     : kLPSPI_ClockPolarityActiveHigh;
-	master_config.cpha = (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA)
-				     ? kLPSPI_ClockPhaseSecondEdge
-				     : kLPSPI_ClockPhaseFirstEdge;
-	master_config.direction =
-		(spi_cfg->operation & SPI_TRANSFER_LSB) ? kLPSPI_LsbFirst : kLPSPI_MsbFirst;
-	master_config.baudRate = spi_cfg->frequency;
-	master_config.pcsToSckDelayInNanoSec = config->pcs_sck_delay;
-	master_config.lastSckToPcsDelayInNanoSec = config->sck_pcs_delay;
-	master_config.betweenTransferDelayInNanoSec = config->transfer_delay;
-	master_config.whichPcs = spi_cfg->slave + kLPSPI_Pcs0;
-	master_config.pcsActiveHighOrLow = (spi_cfg->operation & SPI_CS_ACTIVE_HIGH)
-				    ? kLPSPI_PcsActiveHigh : kLPSPI_PcsActiveLow;
-	master_config.pinCfg = config->data_pin_config;
-	master_config.dataOutConfig = config->tristate_output ? kLpspiDataOutTristate :
-								kLpspiDataOutRetained;
-
-	LPSPI_MasterInit(base, &master_config, clock_freq);
-	LPSPI_SetDummyData(base, 0);
+	lpspi_basic_config(dev, spi_cfg);
 
-	if (IS_ENABLED(CONFIG_DEBUG)) {
-		base->CR |= LPSPI_CR_DBGEN_MASK;
+	ret = clock_control_get_rate(config->clock_dev, config->clock_subsys, &clock_freq);
+	if (ret) {
+		return ret;
 	}
 
+	lpspi_set_sckdiv(base, spi_cfg->frequency, clock_freq, &prescaler);
+	lpspi_set_delays(dev, clock_freq / TWO_EXP(prescaler));
+
+	base->CR |= LPSPI_CR_MEN_MASK;
+
+	base->TCR = LPSPI_TCR_CPOL(!!(spi_cfg->operation & SPI_MODE_CPOL)) |
+		    LPSPI_TCR_CPHA(!!(spi_cfg->operation & SPI_MODE_CPHA)) |
+		    LPSPI_TCR_LSBF(!!(spi_cfg->operation & SPI_TRANSFER_LSB)) |
+		    LPSPI_TCR_FRAMESZ(SPI_WORD_SIZE_GET(spi_cfg->operation)) |
```

Comment:
The FRAMESZ register field requires "number_of_bits - 1", plus, you probably want to use the clamped `word_size` from line 297 (which you probably wanted to as it's actually unused right now?)

Reply:
man I hate that I messed that up lol